src/core/tileLayer.js

```diff
@@ -920,9 +912,31 @@
 
           // mark the tile as covered
           this._setTileTree(tile);
-        }.bind(this));
+        } else {
+          tile.then(function () {
+            if (tile !== this.cache.get(tile.toString())) {
+              /* If the tile has fallen out of the cache, don't draw it -- it
```

Comment:
Yes.  This can happen, for instance, when you quickly pan a large distance.  We fetch an image at the beginning of the pan, and by the time we fully get the image, the map has been panned to that the tile is no longer being displayed.  We no longer want to keep the tile or image in memory, so we want to discard it.